modules/renter/download.go

```diff
@@ -63,15 +62,17 @@ type (
 		atomicDataReceived uint64
 		downloadComplete   bool
 		downloadErr        error
-		finishedChunks     []bool
+		finishedChunks     map[int]bool
+		offset             uint64
+		length             uint64
 
 		// Timestamp information.
 		completeTime time.Time
 		startTime    time.Time
 
 		// Static information about the file - can be read without a lock.
 		chunkSize         uint64
-		destination       string
+		destination       modules.DownloadWriter
```

Comment:
I think it actually makes sense to transition this here to a per-chunk thing. Basically, a `download` as managed by the core download loop should only be doing one chunk at a time. Should be able to do offset at length within the chunk (so you can download partial chunks), but because we've now got 3 completely different download scenarios (repair, file download, stream download) and they are all going to want to organize their chunks and priorities a bit differently.